migrator.sh

```diff
@@ -293,10 +301,38 @@ pull_images_from_v1() {
   for i in ${FULL_IMAGE_LIST}
   do
     push_pull_image "pull" "${V1_REGISTRY}/${i}"
+    retag_image "${V1_REGISTRY}/${i}" "${V2_REGISTRY}/${i}"
+    push_pull_image "push" "${V2_REGISTRY}/${i}"
   done
   echo -e "${OK} Successully pulled all images from ${V1_REGISTRY} to your local system"
 }
 
+pull_retag_push_remove_image() {
+  GLOBIMAGE="${1}"
+
+  echo -e "\n${INFO} Processing ${GLOBIMAGE}"
+
+  IMAGE_NAME=$(echo ${GLOBIMAGE} | cut -d ':' -f1)
+  TAG=$(echo ${GLOBIMAGE} | cut -d ':' -f2)
+
+  STATUSCODE=$(curl -s -o /dev/null -w "%{http_code}" https://${AUTH_CREDS}@${V2_REGISTRY}/v2/${IMAGE_NAME}/manifests/${TAG})
+  if [[ $STATUSCODE -eq "200" ]]; then
+    echo -e "\n${INFO} Tag exists, skipping ${GLOBIMAGE}"
```

Comment:
In what case would I want to skip pushing an image?  I understand not wanting to pull an image if it already exists in my v2.  There isn't really an accurate way to compare the images in the v1 and v2 without pulling them and attempting to push them to the best of my knowledge.  My thought process here is this: just because a tag exists doesn't necessarily mean that it is an up to date image.  I want all of my images and tags to be migrated to my new registry in the exact same state on a v2 as a v1 and without a guarantee of that, it leaves a potential gap.  In this case, I would rather pull the image and attempt to push it and let the registry take care of whether or not it is up to date.